WireAuthentication/Sources/WireAuthenticationUI/Views/DetermineAuthMethod/SwitchBackendConfirmation.swift

```diff
@@ -0,0 +1,258 @@
+            .minimumScaleFactor(0.8)
+    }
+
+    private var backendDetails: some View {
+        Group {
+            if showFullDetails {
+                fullDetails
+                    .transition(.asymmetric(insertion: .opacity, removal: .opacity))
+            } else {
+                shortDetails
+                    .transition(.asymmetric(insertion: .opacity, removal: .opacity))
+            }
+        }
+        .animation(.default, value: showFullDetails)
+    }
+
+    private var contentHeight: CGFloat {
+        UIScreen.main.bounds.height * 0.4
+    }
+
+    private var fullDetails: some View {
+        ScrollView {
+            VStack(spacing: 16) {
+                Text(Strings.message)
+                    .foregroundStyle(Color.primaryText)
+                    .multilineTextAlignment(.center)
+                    .lineLimit(nil)
+                    .fixedSize(horizontal: false, vertical: true)
+
+                ForEach(items, id: \.title) { model in
+                    itemView(
+                        title: model.title,
+                        value: model.value,
+                        isURL: model.isURL
+                    )
+                }
+            }
+        }
+        .frame(maxHeight: contentHeight)
+    }
+
+    private var shortDetails: some View {
+        ScrollView {
+            VStack(spacing: 16) {
+                Text(Strings.message)
+                    .foregroundStyle(Color.primaryText)
+                    .multilineTextAlignment(.center)
+                    .lineLimit(nil)
+                    .fixedSize(horizontal: false, vertical: true)
+                ForEach(items.prefix(2), id: \.title) { model in
+                    itemView(
+                        title: model.title,
+                        value: model.value,
+                        isURL: model.isURL
+                    )
+                }
+                Button {
+                    withAnimation {
+                        showFullDetails.toggle()
+                    }
+                } label: {
+                    Text(Strings.showDetails)
+                        .font(.textStyle(.body1))
+                }
+                .wireButtonStyle(.link)
+            }
+        }
+        .frame(maxHeight: contentHeight)
+    }
+
+    private func itemView(
+        title: String,
+        value: String,
+        isURL: Bool = false
+    ) -> some View {
+        VStack {
+            Text(title)
+                .foregroundStyle(Color.secondaryText)
+                .lineLimit(nil)
+                .fixedSize(horizontal: false, vertical: true)
+            Text(value)
+                .foregroundStyle(Color.primaryText)
+                .accessibilityTextContentType(isURL ? .fileSystem : .plain)
+        }
+    }
+
+    private var buttons: some View {
+        VStack(spacing: 6) {
+            cancelButton
+            proceedButton
+        }
+    }
+
+    private var cancelButton: some View {
```

Comment:
question (non-blocking): is there any advantage here, to have all methods private?